### PR TITLE
New sequelize

### PR DESCRIPTION
**Pull request by aviehu**

### CODE REVIEW

File: coverage/clover.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (roeeyud) [inline coverage/clover.xml]: remove and gitignore

File: coverage/coverage-final.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{"C:\\Users\\Avieh\\Desktop\\work\\Deploy-Site\\Deployed-site\\lib\\app\\index.js": {"path":"C:\\Users\\Avieh\\Desktop\\work\\Deploy-Site\\Deployed-site\\lib\\app\\index.js","statementMap":{"0":{"start":{"line":7,"column":12},"end":{"line":7,"column":21}},"1":{"start":{"line":9,"column":0},"end":{"line":9,"column":52}},"2":{"start":{"line":10,"column":0},"end":{"line":10,"column":27}},"3":{"start":{"line":11,"column":0},"end":{"line":11,"column":42}},"4":{"start":{"line":12,"column":0},"end":{"line":12,"column":23}},"5":{"start":{"line":13,"column":0},"end":{"line":13,"column":28}},"6":{"start":{"line":14,"column":0},"end":{"line":14,"column":25}},"7":{"start":{"line":16,"column":0},"end":{"line":16,"column":21}}},"fnMap":{},"branchMap":{"0":{"loc":{"start":{"line":11,"column":16},"end":{"line":11,"column":40}},"type":"binary-expr","locations":[{"start":{"line":11,"column":16},"end":{"line":11,"column":32}},{"start":{"line":11,"column":36},"end":{"line":11,"column":40}}],"line":11}},"s":{"0":1,"1":1,"2":1,"3":1,"4":1,"5":1,"6":1,"7":1},"f":{},"b":{"0":[1,1]},"_coverageSchema":"332fd63041d2c1bcb487cc26dd0d5f7d97098a6c","hash":"7e397d3dd6fbb9efa15e021d009082c74320eac2"}
```
Comment (roeeyud) [inline coverage/coverage-final.json]: remove! + ignore

File: coverage/lcov-report/app/index.html [hunk 3]
```diff
@@ -0,0 +1,97 @@
+<!doctype html>
```
Comment (roeeyud) [inline coverage/lcov-report/app/index.html]: same

File: coverage/lcov-report/app/index.js.html [hunk 4]
```diff
@@ -0,0 +1,117 @@
+<!doctype html>
```
Comment (roeeyud) [inline coverage/lcov-report/app/index.js.html]: same

File: coverage/lcov-report/base.css [hunk 5]
```diff
@@ -0,0 +1,223 @@
+body, html {
```
Comment (roeeyud) [inline coverage/lcov-report/base.css]: same

File: coverage/lcov-report/block-navigation.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
+var jumpToCode = (function init () {
```
Comment (roeeyud) [inline coverage/lcov-report/block-navigation.js]: same

File: coverage/lcov-report/handlers/getIndex.js.html [hunk 7]
```diff
@@ -0,0 +1,90 @@
+<!doctype html>
```
Comment (roeeyud) [inline coverage/lcov-report/handlers/getIndex.js.html]: same

File: coverage/lcov-report/handlers/getUsers.js.html [hunk 8]
```diff
@@ -0,0 +1,105 @@
+<!doctype html>
```
Comment (roeeyud) [inline coverage/lcov-report/handlers/getUsers.js.html]: same

File: coverage/lcov-report/handlers/index.html [hunk 9]
```diff
@@ -0,0 +1,123 @@
+<!doctype html>
```
Comment (roeeyud) [inline coverage/lcov-report/handlers/index.html]: same

File: coverage/lcov-report/handlers/postUsers.js.html [hunk 10]
```diff
@@ -0,0 +1,96 @@
+<!doctype html>
```
Comment (roeeyud) [inline coverage/lcov-report/handlers/postUsers.js.html]: same

File: coverage/lcov-report/index.html [hunk 11]
```diff
@@ -0,0 +1,123 @@
+<!doctype html>
```
Comment (roeeyud) [inline coverage/lcov-report/index.html]: same

File: coverage/lcov-report/models/connection.js.html [hunk 12]
```diff
@@ -0,0 +1,114 @@
+<!doctype html>
```
Comment (roeeyud) [inline coverage/lcov-report/models/connection.js.html]: same

File: coverage/lcov-report/models/index.html [hunk 13]
```diff
@@ -0,0 +1,110 @@
+<!doctype html>
```
Comment (roeeyud) [inline coverage/lcov-report/models/index.html]: same

File: coverage/lcov-report/models/users.js.html [hunk 14]
```diff
@@ -0,0 +1,108 @@
+<!doctype html>
```
Comment (roeeyud) [inline coverage/lcov-report/models/users.js.html]: same

File: coverage/lcov-report/sorter.js [hunk 15]
```diff
@@ -0,0 +1,158 @@
+var addSorting = (function () {
```
Comment (roeeyud) [inline coverage/lcov-report/sorter.js]: remove

File: coverage/lcov.info [hunk 16]
```diff
@@ -0,0 +1,81 @@
+TN:
```
Comment (roeeyud) [inline coverage/lcov.info]: remove

File: lib/app/index.js [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+import bodyParser from 'body-parser';
+import getIndex from '../handlers/getIndex';
+import getUsers from '../handlers/getUsers';
+import postUsers from '../handlers/postUsers';
+
+const app = express();
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+app.set('port', process.env.PORT || 5000);
```
Comment (roeeyud) [inline lib/app/index.js]: do you understand this line `process.env.PORT `

File: lib/handlers/getUsers.test.js [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+import getUsers from './getUsers';
+import users from '../models/users';
+
+describe('getUsers', () => {
+  beforeEach(async () => {
+    await users.sync();
+  });
+
+  afterEach(async () => {
+    const usersIns = await users.findAll();
```
Comment (roeeyud) [inline lib/handlers/getUsers.test.js]: Find a better way to empty table

File: lib/handlers/getUsers.test.js [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+  it('should return the complete list of users in DB', async () => {
+    const send = jest.fn();
+    const data = [
+      {
+        id: 1,
+        userName: 'Aviehu',
+        age: '24',
+      },
+    ];
+    await users.sync()
```
Comment (roeeyud) [inline lib/handlers/getUsers.test.js]: do you need to sync again?

File: lib/handlers/postUsers.js [hunk 20]
```diff
@@ -0,0 +1,9 @@
+import connection from '../models/connection';
+import users from '../models/users';
+
+async function postUsers(req) {
+  await connection.sync();
```
Comment (roeeyud) [inline lib/handlers/postUsers.js]: you are using sync inconsistently.
Some handlers call sync and some don't... 
choose a single concept

File: lib/models/users.js [hunk 21]
```diff
@@ -0,0 +1,13 @@
+import Sequelize from 'sequelize';
+import connection from './connection';
+
+const users = connection.define('test', {
+  userName: Sequelize.STRING,
+  age: Sequelize.STRING,
+}, {
+  timestamps: false,
```
Comment (roeeyud) [inline lib/models/users.js]: What does this line do?

File: lib/models/connection.js [hunk 22]
```diff
@@ -0,0 +1,15 @@
+import Sequelize from 'sequelize';
+
+const connection = new Sequelize('database', 'postgres', 'aqswde', {
+  host: 'localhost',
+  dialect: 'sqlite',
+  operatorsAliases: false,
+  pool: {
```
Comment (roeeyud) [inline lib/models/connection.js]: Do you understand the pool property?

File: package.json [hunk 23]
```diff
@@ -31,6 +42,10 @@
     "eslint-plugin-import": "^2.14.0",
     "eslint-plugin-node": "^7.0.1",
     "eslint-plugin-promise": "^4.0.0",
-    "eslint-plugin-standard": "^3.1.0"
+    "eslint-plugin-standard": "^3.1.0",
+    "jest": "^23.5.0",
+    "regenerator-runtime": "^0.12.1",
+    "superagent": "^3.8.3",
+    "supertest": "^3.1.0"
```
Comment (roeeyud) [inline package.json]: I'll give you one last chance to clean-up package JSON from any unused packages.

File: .circleci/jest.config.js [hunk 24]
```diff
@@ -0,0 +1,3 @@
+{
+    reporters: ["default", "jest-junit"]
```
[the remark this answers is not in the document]
Comment (aviehu) [reply .circleci/jest.config.js]: yep ill remove the file

File: lib/models/users.js (hunk 21 above)
Comment (aviehu) [reply lib/models/users.js]: creates a row in the table which has "userName" and "age" both strings
timestamps: false means it does not contain "createdAt" and "updatedAt"

File: package.json (hunk 23 above)
Comment (aviehu) [reply package.json]: ok if i do that and run yarn install will it remove all the unused one from node.moudels?

File: lib/models/connection.js (hunk 22 above)
Comment (aviehu) [reply lib/models/connection.js]: not really to be honest

File: lib/handlers/getUsers.test.js (hunk 19 above)
Comment (aviehu) [reply lib/handlers/getUsers.test.js]: no i didnt notice i still had it

File: lib/handlers/postUsers.js (hunk 20 above)
Comment (aviehu) [reply lib/handlers/postUsers.js]: what do you mean by choose a single concept? looking at the code .sync just shouldn't be there

File: lib/models/users.js (hunk 21 above)
Comment (roeeyud) [reply lib/models/users.js]: return the timestamps pls

File: lib/models/connection.js (hunk 22 above)
Comment (roeeyud) [reply lib/models/connection.js]: Maybe you should... just be sure that these lines should be here

File: lib/models/users.js (hunk 21 above)
Comment (aviehu) [reply lib/models/users.js]: no problem

File: lib/app/index.js (hunk 17 above)
Comment (aviehu) [reply lib/app/index.js]: app.set('port', process.env.PORT || 5000); -> either set the port 5000 or use the process.env.PORT that heroku (or other similar services) determines

File: lib/handlers/getUsers.test.js (hunk 18 above)
Comment (aviehu) [reply lib/handlers/getUsers.test.js]: no problem, but why not destroy?

Comment (aviehu) [reply lib/handlers/getUsers.test.js]: wait is it ok if i use destroy on the model and not the instance

File: .gitignore [hunk 25]
```diff
@@ -1,2 +1,2 @@
 /node_modules
-/package-lock.json
\ No newline at end of file
+/.vscode
```
Comment (roeeyud) [inline .gitignore]: add coverage

**Pull request merged**